### PR TITLE
feat(unified-share-modal): add property to limit the number of contacts

### PR DESCRIPTION
**Pull request by stevegcoder**
Add property to limit the number of contact pills that can be created at one time.
Remove reset of error on contact input.  This preserves the contact limit error.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @stevegcoder has signed the CLA. Thanks for the pull request!

**Comment by priyajeet** [comment]
Is it a breaking change? As in the consumer will have to make changes otherwise it will break?

**Comment by cweeii** [comment]
@priyajeet This shouldn't be breaking since contactLimit is an optional prop.

File: src/features/unified-share-modal/EmailForm.js [hunk 1]
```diff
@@ -81,21 +82,37 @@ class EmailForm extends React.Component<Props, State> {
...
-        updateSelectedContacts([...this.props.selectedContacts, ...contacts]);
+        updateSelectedContacts([...selectedContacts, ...contacts]);
+
+        let contactsFieldError = '';
+        if (contactLimit !== undefined && selectedContacts.length + contacts.length > contactLimit) {
+            contactsFieldError = intl.formatMessage(messages.contactsExceedLimitError, {
+                maxContacts: contactLimit,
+            });
+        }
+        this.setState({ contactsFieldError });
```
Comment (cweeii) [inline src/features/unified-share-modal/EmailForm.js]: It seems like you're doing this calculation and setting state 3 times in handleContactAdd, handleContactRemove, and handleSubmit.  Can we move it to componentDidUpdate?

**Comment by stevegcoder** [comment]
@priyajeet I put breaking change because it doesn't clear the error when typing into the input.  This is more of a change/break in functionality, so that's why I have it tagged.  I can remove Breaking change.

File: src/features/unified-share-modal/EmailForm.js [hunk 2]
```diff
@@ -203,6 +202,23 @@ class EmailForm extends React.Component<Props, State> {
         return emailValidator(text);
     };
 
+    componentDidUpdate(prevProps: Props, prevState: State) {
```
[the remark this answers is not in the document]
Comment (stevegcoder) [reply src/features/unified-share-modal/EmailForm.js]: Hey, I added a check to disable the button when there's an error.  Also I moved the check from componentDidUpdate to onContactAdd/Remove and submit to be more explicit.

File: i18n/en-US.properties [hunk 3]
```diff
@@ -1168,6 +1168,8 @@ boxui.unifiedShare.collaboration.userCollabText = User
 boxui.unifiedShare.collaboratorListTitle = People in '{itemName}'
 # This string is displayed as tooltip on hovering over expire icon for collab
 boxui.unifiedShare.collaborators.expirationTooltipClickableText = Access expires on {date}. Click for details.
+# Error message when more than the maximum number of contacts is entered
+boxui.unifiedShare.contactsExceedLimitError = Oops! You've reached the maximum number of collaborators that can be added at once ({maxContacts} collaborators). Please try again by splitting your invitation into batches.
```
Comment (cweeii) [inline i18n/en-US.properties]: Can we change the wording to `Oops!  The maximum number of collaborators that can be added at once is {maxContacts} collaborators.  Please try again by splitting your invitations into batches.`.  I think it's more easy to localize than having something in parentheses.

**Pull request merged**